bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/CreateAdvBuilder.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *pa
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.bookkeeper.client.AsyncCallback.CreateCallback;
+import org.apache.bookkeeper.client.BKException;
+
+/**
+ * Builder-style interface to create new ledgers
+ */
+public interface CreateAdvBuilder {
```

Comment:
why have a separate builder for this? Surely the ledger once returned behaves the same? I'd just put withLedgerId on the normal CreateBuilder, and return a different ledger object based on whether it's used.

Reply:
@ivankelly 
LedgerHandler has different semantics than LedgerHandlerAdv (I you are using Adv you MUST manage entryId manually and using non-Adv you CANNOT manage entryId manually)
and so we need a more clear API.
So we have WriteHandler and WriteAdvHandler, which have only the methods which make sense to have.
We need a new interface to have a strongly typed API
the other options is to have
CreateBuilder {
         WriteHandler create()
         WriteAdvHandler createAdv();
}
additionaly for "Adv" you can force a specific ledgerId